perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -585,6 +586,37 @@ def DisallowAllPorts(self):
     """Closes all ports on the firewall."""
     for firewall_rule in six.itervalues(self.firewall_rules):
       firewall_rule.Delete()
+    for firewall_rule in six.itervalues(self.firewall_icmp_rules):
+      firewall_rule.Delete()
+
+  def AllowIcmp(self, vm):
+    """Opens the ICMP protocol on the firewall.
+
+    Args:
+      vm: The BaseVirtualMachine object to open the ICMP protocol for.
+    """
+    if vm.is_static:
+      return
+    with self._lock:
+      if vm.cidr:  # Allow multiple networks per zone.
+        cidr_string = network.BaseNetwork.FormatCidrString(vm.cidr)
+        firewall_name = ('perfkit-firewall-icmp-%s-%s' %
+                         (cidr_string, FLAGS.run_uri))
+        key = (vm.project, vm.cidr)
```

Comment:
It looks like GCEFirewall doesn't use vm.cidr but vm.network.network_resource.name when generating a rule.
I take it there's a 1:1 mapping between cidr and the network name?

Reply:
Yes, I tried to implement this in the same way as in the AllowPort method.

If the cidr string is 10-0-1-0-24, then the network resource name would be something like
pkb-network-multi-10-0-1-0-24-(run-uri).

I have also tested and confirmed that this works with all of the network configs in gcp_network_test